packages/client/hmi-client/src/components/responsive-matrix/matrix.vue

```diff
@@ -0,0 +1,902 @@
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	async beforeMount() {
+		this.processData();
+		this.processLabels();
+		this.processActiveCells();
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// mounted                                                                      //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	async mounted() {
+		const matrix = this.$refs.matrix as HTMLElement;
+		const matrixContainer = this.$refs.matrix_container as HTMLElement;
+
+		// ///////////////////////////////////////////////////////////////////////////////
+
+		// initialize screen height/width
+		this.screenHeight = matrix.offsetHeight;
+		this.screenWidth = matrix.offsetWidth;
+
+		// start the resize observer
+		this.resizeObserver = new ResizeObserver((entries) => {
+			this.screenHeight = entries[0].contentRect.height;
+			this.screenWidth = entries[0].contentRect.width;
+		});
+		this.resizeObserver.observe(matrixContainer);
+
+		// start event listeners
+		window.addEventListener('keydown', this.handleKey);
+		window.addEventListener('keyup', this.handleKey);
+		matrixContainer.addEventListener('wheel', this.handleScroll);
+
+		// ///////////////////////////////////////////////////////////////////////////////
+
+		// initialize pixi.js
+		const app = new Application({
+			resizeTo: matrixContainer,
+			backgroundColor: this.backgroundColor
+		});
+		matrix.appendChild(app.view as unknown as Node);
+
+		// create viewport
+		const screenHeight = this.screenHeight;
+		const screenWidth = this.screenWidth;
+		const cellAspectRatio = 1 || this.microRowArray.length / this.microColArray.length;
+		const quadAspectRatio = this.screenAspectRatio * cellAspectRatio; // height / width
+		const vertexPosition =
+			quadAspectRatio > 1 // is height long?
+				? [
+						-1 / quadAspectRatio,
+						-1, // height long
+						1 / quadAspectRatio,
+						-1, // x, y
+						1 / quadAspectRatio,
+						1,
+						-1 / quadAspectRatio,
+						1
+				  ]
+				: [
+						-1,
+						-1 * quadAspectRatio, // width long
+						1,
+						-1 * quadAspectRatio, // x, y
+						1,
+						1 * quadAspectRatio,
+						-1,
+						1 * quadAspectRatio
+				  ];
+		this.worldHeight = vertexPosition[5] - vertexPosition[3];
+		this.worldWidth = vertexPosition[2] - vertexPosition[0];
+		(this as any).$viewport = new Viewport({
+			screenWidth: this.screenWidth,
+			screenHeight: this.screenHeight,
+			worldWidth: this.worldWidth,
+			worldHeight: this.worldHeight,
+			divWheel: matrix as HTMLElement,
+			interaction: app.renderer.plugins.interaction // the interaction module is important for wheel to work properly when renderer.view is placed or scaled
+		});
```

Comment:
i dont like this but i did this specifically to avoid reactivity (which breaks everything). lemme know if there are any other solutions you can think of im open to suggestions.

Reply:
Hmmm, so your code is kind of in a quirky phase here because it is written in a Vue-2 style, but the rest of the application is using Vue-3.

I think you can declare non-reactive variables using setup()
```
setup() {
  const $viewport = ...

  return { $viewport }
}
```

Going to ping @bencodrington here because he knows a bit more about this than I ...

Reply:
Yep, @mwdchang's suggestion above checks out 👍 variables returned from `setup()` are non-reactive unless you explicitly specify they should be using `ref()`, `reactive()`, or `computed()`.

Assuming you want all this to still run after the component has mounted, you'd do something like:
```ts
setup() {
  let $viewPort: ViewPort | null = null;
  onMounted(() => {
    $viewPort = new ViewPort(...)
  }
  return { $viewPort };
}
```

The [docs page](https://vuejs.org/api/composition-api-setup.html#setup-context) is here, let me know if there's anything else I can clear up for ya.

Reply:
thanks! will do it this way